lib/std/fs/Dir.zig

```diff
@@ -566,59 +629,11 @@ pub const Iterator = switch (builtin.os.tag) {
     else => @compileError("unimplemented"),
 };
 
-pub fn iterate(self: Dir) Iterator {
-    return self.iterateImpl(true);
-}
-
-/// Like `iterate`, but will not reset the directory cursor before the first
-/// iteration. This should only be used in cases where it is known that the
-/// `Dir` has not had its cursor modified yet (e.g. it was just opened).
-pub fn iterateAssumeFirstIteration(self: Dir) Iterator {
```

Comment:
I don't know (first PR :see_no_evil: )

Zig sometimes marks functions as `Deprecated`. I could add `iterateAssumeFirstIteration` again and mark it as deprecated, but this won't work for `iterator()`. Therefore I also removed `iterateAssumeFirstIteration` rather than trying to deprecate it.

Reply:
I dug into this a bit more ... the example I'm remembering is that on #18271 it has a "breaking" label.  I'm not entirely sure if that applies here, but it seems likely.

Reply:
somebody with proper Permissions has to add this ;)